vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -78,69 +132,176 @@
+                datepicker.i18n.formatDate = tryCatchWrapper(formatDate);
 
-                datepicker.i18n.parseDate = tryCatchWrapper(function (dateString) {
+                /**
+                 * @param {string} dateString
+                 * @returns {DateHash|boolean|undefined}
+                 */
+                let _parseDate = function (dateString) {
                     dateString = cleanString(dateString);
 
-                    if (dateString.length == 0) {
+                    if (dateString.length === 0) {
                         return;
                     }
 
                     let match = dateString.match(datepicker.$connector.regex);
-                    if (match && match.length == 4) {
+                    if (match && match.length === 4) {
+                        /** @type {number} */
+                        let day, month, year
+                        /** @type {string} */
+                        let dayMatch, monthMatch, yearMatch;
+
                         for (let i = 1; i < 4; i++) {
-                            datepicker.$connector.parts[i-1].value = parseInt(match[i]);
+                            const part = parts[i-1];
+                            const matchValue = match[i];
+                            part.value = parseInt(matchValue);
+                            if (part === dayPart) {
+                                day = part.value;
+                                dayMatch = matchValue;
+                            } else if (part === monthPart) {
+                                month = part.value - 1;
+                                monthMatch = matchValue;
+                            } else if (part === yearPart) {
+                                year = part.value;
+                                yearMatch = matchValue;
+                            }
                         }
-                        return {
-                            day: datepicker.$connector.dayPart.value,
-                            month: datepicker.$connector.monthPart.value - 1,
-                            year: datepicker.$connector.yearPart.value
-                        };
-                    }  else {
+                        return {day, month, year, dayMatch, monthMatch, yearMatch};
+                    } else {
                         return false;
                     }
-                });
+                };
+                _parseDate = tryCatchWrapper(_parseDate);
+
+                /**
+                 * @param {string} dateString
+                 * @returns {DateHash|boolean|undefined}
+                 */
+                const parseDate = function (dateString) {
+                    const date = _parseDate(dateString);
+                    if (!date) {
+                        return date;
+                    }
+                    if (date.yearMatch.length < 3 && date.year >= 0) {
+                        date.year += date.year < 50 ? 2000 : 1900;
+                    }
```

Comment:
This logic was copied from the web component for consistent behaviour. Two digit year (when not prefixed with zeros) is converted to either `20xx` or `19xx`.